library/pod-security-policy/seccomp/template.yaml

```diff
@@ -35,52 +35,220 @@ spec:
+        }
+
+        get_message(profile, file, name, location, allowed_profiles) = message {
+        	profile == "Localhost"
+        	message := sprintf("Seccomp profile '%v' with file '%v' is not allowed for container '%v'. Found at: %v. Allowed profiles: %v", [profile, file, name, location, allowed_profiles])
+        }
+
+        input_wildcard_allowed_profiles {
+        	input.parameters.allowedProfiles[_] == "*"
+        }
+
+        input_wildcard_allowed_files {
+        	input.parameters.allowedLocalhostFiles[_] == "*"
+        }
+
+        input_wildcard_allowed_files {
+        	"localhost/*" == input.parameters.allowedProfiles[_]
+        }
+
+        # Simple allowed Profiles
+        allowed_profile(profile, file, allowed) {
+        	not startswith(lower(profile), "localhost")
+        	profile == allowed[_]
+        }
+
+        # seccomp Localhost without wildcard
+        allowed_profile(profile, file, allowed) {
+        	profile == "Localhost"
+        	not input_wildcard_allowed_files
+        	profile == allowed[_]
+        	allowed_files := {x | x := object.get(input.parameters, "allowedLocalhostFiles", [])[_]} | get_annotation_localhost_files
+        	file == allowed_files[_]
         }
 
-        input_wildcard_allowed(metadata) {
-            input.parameters.allowedProfiles[_] == "*"
+        # seccomp Localhost with wildcard
+        allowed_profile(profile, file, allowed) {
+        	profile == "Localhost"
+        	input_wildcard_allowed_files
+        	profile == allowed[_]
         }
 
-        input_container_allowed(metadata, container) {
-            not get_container_profile(metadata, container)
-            metadata.annotations["seccomp.security.alpha.kubernetes.io/pod"] == input.parameters.allowedProfiles[_]
+        # annotation localhost with wildcard
+        allowed_profile(profile, file, allowed) {
+        	"localhost/*" == allowed[_]
+        	startswith(profile, "localhost/")
         }
 
-        input_container_allowed(metadata, container) {
-            profile := get_container_profile(metadata, container)
-            profile == input.parameters.allowedProfiles[_]
+        # annotation localhost without wildcard
+        allowed_profile(profile, file, allowed) {
+        	startswith(profile, "localhost/")
+        	profile == allowed[_]
         }
 
-        get_container_profile(metadata, container) = profile {
-            value := metadata.annotations[key]
-            startswith(key, "container.seccomp.security.alpha.kubernetes.io/")
-            [prefix, name] := split(key, "/")
-            name == container.name
-            profile = value
+        # Localhost files from annotation scheme
+        get_annotation_localhost_files[file] {
+        	profile := input.parameters.allowedProfiles[_]
+        	startswith(profile, "localhost/")
+        	file := split(profile, "/")[1]
```

Comment:
to guard against file names in subdirectories (not sure if that's possible), do we want to trim the `localhost/` prefix instead?

Reply:
Yeah I think that would be better, I was thinking about this earlier as a potential problem if there can be another `/` in the path.